### ISSUE TITLE
Unsupervised Representation Learning with Deep Convolutional Generative Adversarial Networks

### ISSUE THREAD

**Issue by reyllama** [reyllama/papers#6]
## Authors

Alec Radford, Luke Metz, Soumith Chintala

## Tags

- Generative Adversarial Networks

- Computer Vision

- Unsupervised Learning

### 1. Introduction

- The advent of Convolutional Neural Network brought great success to supervised learning of images, but the unsupervised learning remained less blessed.

- GAN, introduced in 2014 by Goodfellow et al, provides a good alternative for learning intermediate representations of image data under unsupervised settings.

- In the introduction part, the authors have stated their share of contributions:

> - Suggest and evaluate constraints on DCGAN architecture to make it stable.

> - Demonstrate effectiveness of DCGAN discriminator as image classifier

> - Empirically show that GAN filters reflect representations of certain images

> - Come up with Generator with vector arithmetic qualities that enable easy manipulation of generated samples

### 2. Related Work

- Previous works on unsupervised representation learning mostly employed traditional clustering techniques or auto-encoders. 

- Generative image models can be classified in two groups: parametric and non-parametric.

- Parametric models such as GAN and recurrent network approach displayed some success, but were not satisfactory in terms of resolution/quality of the generated image.

- Interpretability of CNN can be achieved by two crucial methods:

> (1) Finding the purpose of each filters by deconvolution and filtering the maximal activation

<br>

> (2) Searching for the ideal image that activates certain filters by using gradient descent on the input sample

<br>

### 3. Approach and Model Architecture

- Architectural guidelines for stable DCGAN:

> (1) Replace pooling layers with strided (D) / fractional strided (G) convolutions

<br>

> (2) Employ batch normalization to help gradient flow in deeper models

<br>

> (3) Remove Dense Layers

<br>

> (4) Activation for Generator : Tanh for output / ReLU for the rest

<br>

> (5) Activation for Discriminator : LeakyReLU for all

<br>

### 4. Details of Adversarial Training

- Only preprocessing being **scaling** training images to [-1,1] for tanh range

- Mini-batch **Stochastic Gradient Descent(SGD)** with batch size 128

- All weights initialized ~ N(0, 0.02^2)

- LeakyReLU weak side slope set to 0.2 and use of Adam Optimizer

- **Learning rate** 0.001 too high, replaced to 0.0002

![image](https://user-images.githubusercontent.com/46308407/78504166-47236580-77a6-11ea-9288-1d858f0a434e.png)

- Large-scale Scene Understanding (LSUN), Randomly scraped Face images and natural images from Imagenet-1K was used.

### 5. Empirical Validations of DCGAN Capabilities

- DCGAN showed impressive performance as feature extractor, scoring higher than any other K-means models in CIFAR-10 classification problem when used as baseline model.

> Robustness across domains can be argued since DCGAN was never trained on CIFAR-10.

![image](https://user-images.githubusercontent.com/46308407/78504513-bf8b2600-77a8-11ea-84dc-e726526a1161.png)

<br>

- Undergoing similar procedures for data preparation, DCGAN achieved state-of-the-art performance for StreetView House Numbers dataset classfication. 

> CNN turns out not to be the primary contributor for DCGAN's success, since training a purely supervised CNN with exhaustive hyper-parameter tuning yielded a far inferior result.

![image](https://user-images.githubusercontent.com/46308407/78504519-cd40ab80-77a8-11ea-80b9-0774ed976523.png)

### 6. Investing and Visualizing the Internals of the Networks

 - **Walking in the Latent Space** shows smooth transition in generated images, implying that model has learned relevant representations.

![image](https://user-images.githubusercontent.com/46308407/78504801-4391dd80-77aa-11ea-9552-0b33aed469a8.png)

- Visualizing the Discriminator features reveals it learned a hierarchy of features, which activate on typical objects such as bed, window, etc.

![image](https://user-images.githubusercontent.com/46308407/78504812-4e4c7280-77aa-11ea-99d6-47fb0d986b1a.png)

- Manipulation on Generator Representation were done in two ways: 

> (1) Forcing to forget certain objects

![image](https://user-images.githubusercontent.com/46308407/78504824-599f9e00-77aa-11ea-9a9d-1eea32b1c339.png)

<br>

> (2) Arithmetic vector manipulation

![image](https://user-images.githubusercontent.com/46308407/78504838-69b77d80-77aa-11ea-9121-597fa01388e0.png)

<br>

### 7. Conclusion and Future Work

- This work showed that adversarial net can learn good representations of images for classification and generation.

- However, model instability was detected as model training prolonged: they sometimes collapsed a subset of filters to a single oscillating mode. Further work required.

## Personal Comments

- These days I'm into GAN, but this paper was somewhat deviated from my expectation in that it focused more on representation learning than generative capabilities. 

- I learned GAN can be used not only as generative model, but also as a feature extractor under unsupervised learning fashion.

- The idea of representation and **latent space** intrigues me and puzzles me at the same time. The term **semantic** as well. I think I need to get more familiar with these terminologies which are used very often.

- Manipulation part was very impressive. It's astonishing how visual information forms a hierarchy of representations, and they can be handled arithmetically. I wish it gives insights to how humans perceive visual signals and form meanings.
